server/src/test/java/io/crate/lucene/NumericEqQueryTest.java

```diff
@@ -115,6 +118,29 @@ public void test_numeric_comparisions_with_different_precision() {
         assertThat(convert(col + " <= -1")).isEqualTo(convert(col + " < -0.99"));
     }
 
+    @Test
+    public void test_equals_with_different_precision() {
+        String col = randomBoolean() ? "x" : "y";
+        assertThat(convert(col + " = 1")).isEqualTo(convert(col + " = 1.00"));
+        assertThat(convert(col + " = 1.1")).isEqualTo(convert(col + " = 1.10"));
+        // TODO the cast is required to preserve the precision/scale until NumericEqQuery
```

Comment:
Please see https://github.com/crate/crate/issues/16838, in my opinion it is a separate bug and is less priority as there is a quick workaround.

Reply:
Actually, I do not understand the TODO comment.